app.js

```diff
@@ -27,10 +28,18 @@ let mainWindow = null;
 let previewWindow = null;
 
 function createTourWindow() {
+  // Easier to work with when centering tour
+  // window on primary display.
+  const width = 700;
+  const height = 600;
+  // Get Primary Display (screen / monitor)
+  const primaryDisplay = electron.screen.getPrimaryDisplay();
   // Creating a New Window
   tourWindow = new BrowserWindow({
-    width: 700,
-    height: 600,
+    x: primaryDisplay.bounds.x / 2 - width / 2 + primaryDisplay.bounds.x,
+    y: primaryDisplay.bounds.y / 2 + height / 2 + primaryDisplay.bounds.y,
```

Comment:
Hmm, can you explain what line 39 and line 40 do? 
Wouldn't `primaryDisplay.bounds.x` and `primaryDisplay.bounds.y` be resolved to `0`?

Reply:
On my monitor `primaryDisplay.bounds.x` would be `1920` and `primaryDisplay.bounds.y` would be `1080`. (That would be my top left corner)

Btw I took a second look at the API and I see **screen.getDisplayNearestPoint(point)**. I think this means I can figure out what monitor you have Manta opened on, maybe I could try and attach the Tour window in the center of the used monitor?

Reply:
Is your `primaryDisplay` on the right? I guess 1920x1080 is the screen size of the other one and it's placed on the left, so the `primaryDisplay.bounds.x` is actually `0+1920` and `primaryDisplay.bounds.y` would be `0 + 1080`? 

I'm using only my laptop and got this:

```
primaryDisplay:  { x: 0, y: 0, width: 1920, height: 1200 }
```
What would happen if you use only 1 monitor? 
Would the code still work?

Reply:
You are correct, my primary display is on the right.
```js
x: primaryDisplay.bounds.x / 2 - width / 2 + primaryDisplay.bounds.x,
y: primaryDisplay.bounds.y / 2 + height / 2 + primaryDisplay.bounds.y,
```
`x = ((0 / 2 = 0) - (700 / 2 = 350) + (0)) = -350`
`y = ((0 / 2 = 0) + (600 / 2 = 300) + (0)) = 300`
It will not work :stuck_out_tongue_closed_eyes: 

If I only use:
```js
x: primaryDisplay.bounds.x,
y: primaryDisplay.bounds.y,
```
For me this is 1920, 1080, but for you 0, 0. It will show on correct screen but not in center :disappointed:

Reply:
```js
const primaryDisplay = electron.screen.getPrimaryDisplay();
let x = undefined
let y = undefined

if (primaryDisplay.bounds.x === 0 && primaryDisplay.bounds.y === 0) {
  x = primaryDisplay.bounds.width / 2 - width / 2 
  y = primaryDisplay.bounds.height / 2 - height / 2
} else {
  x = primaryDisplay.bounds.x / 2 - width / 2 + primaryDisplay.bounds.x
  y = primaryDisplay.bounds.y / 2 + height / 2 + primaryDisplay.bounds.y
}
```
If PrimaryDisplay is Monitor 1 we can assume it start at 0, 0, right?
`x = ((1920 /  2 = 960) - (700 / 2 = 350)) = 610`
`y = ((1080 / 2 = 540) - (600  / 2 = 300)) =  240`

If PrimaryDisplay is Monitor 2 or higher:
`x = ((1920 / 2 = 960) - (700 / 2 = 350) + (1920)) = 2530`
`y = ((0 / 2 = 0) - (600 / 2 = 300) + (0)) = 300`
*x and y is 1920, 0 if I have monitors horizontal to each other* - [Monitor1][Monitor2]

**Tests** (The tour window is in center)
Two Monitors: *(ran tests with both vertical and horizontal monitors)*
I made Monitor 1 my primary display and tested, it worked.
I made Monitor 2 my primary display and tested, it worked.

One Monitor:
I unplugged monitor 1 so I only have 1 monitor and tested, it worked.

--------------------
If you have some weird monitor configure, such as if right top corner monitor 1 is linked to bottom left corner monitor 2 than tour window wont be 100% in center, but somewhere close to it :stuck_out_tongue:

Reply:
What if we use this in all cases? 

```js
x = primaryDisplay.bounds.width / 2 - width / 2 
y = primaryDisplay.bounds.height / 2 - height / 2
```
Because what we want is for the `tourWindow` to be in the center of the `primaryDisplay`, regardless if the setup? (Meaning the only thing that it needs to know is the dimensions of the `primaryDisplay` to calculate the position of the `tourWindow`)

Reply:
*For the sake of this example I use the same resolution for all monitors.*
![untitled](https://user-images.githubusercontent.com/13605163/34905674-d811da22-f855-11e7-9933-84f023119d75.jpg)

if I use
```js
x = primaryDisplay.bounds.width / 2 - width / 2 
y = primaryDisplay.bounds.height / 2 - height / 2
```
**x** and **y** will **always** be the same because the resolution is the same on all monitors. This means the tour window will **always** show on monitor 1 regardless whether it is the primary display or not. If one monitor would be a different resolution and it was the primary display, it would still show on monitor 1 but not in center.

Let's assume monitor 5 is the primary display (worst case scenario).
`x = ((1920 / 2 = 960) - (700 / 2 = 350)) = 610`
`y = ((1080 / 2 = 540) - (600 / 2 = 300)) = 240`
*This is x, y range of Monitor 1*

------------------------

```js
x = primaryDisplay.bounds.x / 2 - width / 2 + primaryDisplay.bounds.x
y = primaryDisplay.bounds.y / 2 + height / 2 + primaryDisplay.bounds.y
```
`x = ((3840 / 2 = 1920) - (700 / 2 = 350) + (3840)) = 5410`
`y = ((1080 / 2 = 540) + (600 / 2 = 300) + (1080)) = 1920`
This would be within the **x, y** range of monitor 5 *(3840, 1080 to 5760, 2160)*.
But this is **still wrong** because I take half of x and y coordinations which does not have the same scaling as the screen resolution.

------------------------

I believe this should work.
```js
x = primaryDisplay.bounds.width / 2 - width / 2 + primaryDisplay.bounds.x
y = primaryDisplay.bounds.height / 2 + height / 2 + primaryDisplay.bounds.y
```
`x = ((1920 / 2 = 960) - (700 / 2 = 350) + (3840)) = 4450`
`y = ((1080 / 2 = 540) + (600 / 2 = 300) + (1080)) = 1920`
This would be within the **x, y** range of monitor 5 *(3840, 1080 to 5760, 2160)*

I dont have more monitors to test this theory on :stuck_out_tongue: I need someone smart to tell me whether I am wrong or not :slightly_smiling_face: